crates/notedeck_columns/src/ui/settings.rs

```diff
@@ -26,405 +97,440 @@ impl SettingsAction {
-                        ui.label(
-                            RichText::new(tr!(
-                                            self.i18n,
-                                            "Others",
-                                         "Label for others settings section"
-                                        ))
-                                .text_style(NotedeckTextStyle::Body.text_style()),
-                        );
-                        ui.separator();
-                        ui.vertical(|ui| {
-                            ui.spacing_mut().item_spacing = vec2(10.0, 10.0);
-
-                            ui.horizontal_wrapped(|ui| {
-                                ui.label(
-                                    RichText::new(
-                                    tr!(
-                                            self.i18n,
-                                            "Show source client",
-                                            "Label for Show source client, others settings section"
-                                        ))
-                                        .text_style(NotedeckTextStyle::Small.text_style()),
-                                );
-
-                                for option in [
-                                    ShowNoteClientOptions::Hide,
-                                    ShowNoteClientOptions::Top,
-                                    ShowNoteClientOptions::Bottom,
-                                ] {
-                                    let label = option.clone().to_string();
-
-                                    if ui
-                                        .selectable_value(
-                                            self.show_note_client,
-                                            option,
-                                            RichText::new(label)
-                                                .text_style(NotedeckTextStyle::Small.text_style()),
-                                        )
-                                        .changed()
-                                    {
-                                        action = Some(SettingsAction::SetShowNoteClient(option));
-                                    }
-                                }
-                            });
-                        });
+            ui.horizontal(|ui| {
+                ui.label(small_richtext(
+                    self.i18n,
+                    "Language:",
+                    "Label for language, Appearance settings section",
+                ));
+
+                let available_locales: Vec<_> =
+                    self.i18n.get_available_locales().iter().cloned().collect();
+
+                ComboBox::from_label("")
+                    .selected_text(tr!(
+                        self.i18n,
+                        &self.settings.locale.clone(),
+                        &format!("Display name for {} language", self.settings.locale.clone())
+                    ))
```

Comment:
The `tr!` macro is not meant to take in variable strings and comments. They're meant to be static strings. `scripts/export_source_strings.py` scans through the Rust code to extract the strings we need translated -- the code is not compiled by the Python script.

Also, the main label should be the non-translated display name of the language in the language itself (rather than the current language). It shouldn't be localized, so we shouldn't be using the `tr!` macro here.

We may optionally choose to display the localized name of the language in the current language, just so that the user has some awareness of what other languages are supported, but it's not strictly necessary.